itests/eth_filter_test.go

```diff
@@ -622,6 +637,144 @@ func TestTxReceiptBloom(t *testing.T) {
 	require.Equal(t, 6, bitsSet)
 }
 
+func TestMultipleEvents(t *testing.T) {
+	blockTime := 500 * time.Millisecond
+	client, _, ens := kit.EnsembleMinimal(t, kit.MockProofs(), kit.ThroughRPC())
+
+	ens.InterconnectAll().BeginMining(blockTime)
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+
+	// create a new Ethereum account
+	key, ethAddr, deployer := client.EVM().NewAccount()
+	// send some funds to the f410 address
+	kit.SendFunds(ctx, t, client, deployer, types.FromFil(10))
+
+	// DEPLOY CONTRACT
+	tx := deployContract(ctx, t, client, ethAddr, "./contracts/MultipleEvents.hex")
+
+	client.EVM().SignTransaction(tx, key.PrivateKey)
+	hash := client.EVM().SubmitTransaction(ctx, tx)
+
+	receipt, err := client.EVM().WaitTransaction(ctx, hash)
+	require.NoError(t, err)
+	require.NotNil(t, receipt)
+	require.EqualValues(t, ethtypes.EthUint64(0x1), receipt.Status)
+
+	// invoke function to emit four events
+	contractAddr := client.EVM().ComputeContractAddress(ethAddr, 0)
+
+	// https://abi.hashex.org/
+	params4Events, err := hex.DecodeString("98e8da00")
+	require.NoError(t, err)
+
+	params3Events, err := hex.DecodeString("05734db70000000000000000000000001cd1eeecac00fe01f9d11803e48c15c478fe1d22000000000000000000000000000000000000000000000000000000000000000b")
+	require.NoError(t, err)
+
+	gasParams, err := json.Marshal(ethtypes.EthEstimateGasParams{Tx: ethtypes.EthCall{
+		From: &ethAddr,
+		To:   &contractAddr,
+		Data: params4Events,
+	}})
+	require.NoError(t, err)
+
+	gaslimit, err := client.EthEstimateGas(ctx, gasParams)
+	require.NoError(t, err)
+
+	maxPriorityFeePerGas, err := client.EthMaxPriorityFeePerGas(ctx)
+	require.NoError(t, err)
+
+	paramsArray := [][]byte{params4Events, params3Events, params3Events}
+
+	var receipts []*api.EthTxReceipt
+	var hashes []ethtypes.EthHash
+
+	nonce := 1
```

Comment:
let's use `nonce` to cap the maximum number of loops, it could be 10 or something largish, but I think we should cap it rather than relying on a 10m timeout; `require.LessThan(t, nonce, 10, "tried too many times to land messages in same tipset")`

Reply:
Done.